File: src/main/java/nl/joelchrist/shorty/exceptions/EntityNotFoundException.java
package nl.joelchrist.shorty.exceptions;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

@ResponseStatus(HttpStatus.NOT_FOUND)
public class EntityNotFoundException extends RuntimeException {
    private final Class clazz;

    public EntityNotFoundException(Class clazz, String criteria) {
        super(String.format("Could't find %s with criteria '%s'", clazz, criteria));
        this.clazz = clazz;
    }
}

File: src/main/kotlin/nl/joelchrist/shorty/exceptions/EntityNotFoundException.kt
package nl.joelchrist.shorty.exceptions

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus
import kotlin.reflect.KClass

@ResponseStatus(value = HttpStatus.NOT_FOUND, reason = "Couldn't find entity with this criteria")
class EntityNotFoundException(entity: KClass<*>, criteria: String) : RuntimeException("Couldn't find $entity with criteria '$criteria'")